src/zeebe/zb/ZBWorker.ts

```diff
@@ -20,11 +20,11 @@ export class ZBWorker<
 		super(config)
 	}
 
-	protected handleJobs(
+	protected async handleJobs(
 		jobs: ZB.Job<WorkerInputVariables, CustomHeaderShape>[]
 	) {
 		// Call task handler for each new job
-		jobs.forEach(async (job) => this.handleJob(job))
+		await Promise.all(jobs.map((job) => this.handleJob(job)))
```

Comment:
So, at the moment the job handlers run in parallel using the Node event loop. I'm looking at the synchronization changes here and below, and trying to work out the motivation. Was this trying to lock down the source of the unhandled exception by making operations synchronous and explicitly catching to get a debug output closer to the source?